### PR TITLE
[Enhancement](catalog) paimon catalog 

### PR DESCRIPTION
**Pull request by yuxuan-luo**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/scan/paimon_reader.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        } else {
+            required_fields << "," << field;
+            columns_types << "#" << type;
+        }
+        index++;
+    }
+    std::map<String, String> params = {{"mock_rows", "10240"},
+                                       {"required_fields", required_fields.str()},
+                                       {"columns_types", columns_types.str()},
+                                       {"hive.metastore.uris", "thrift://172.16.65.16:7004"},
```
Comment (wsjz) [inline be/src/vec/exec/scan/paimon_reader.cpp:61]: why use domain ip and here?

Comment (yuxuan-luo) [reply be/src/vec/exec/scan/paimon_reader.cpp:61]: Only for test

File: be/src/io/cache/block/block_file_cache_settings.h [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 #include "io/cache/block/block_file_cache_fwd.h"
 
+#include <array>
```
Comment (wsjz) [inline be/src/io/cache/block/block_file_cache_settings.h:27]: Where is the include used

Comment (yuxuan-luo) [reply be/src/io/cache/block/block_file_cache_settings.h:27]: https://github.com/apache/doris/pull/19199/files like this

File: be/src/vec/exec/scan/paimon_reader.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                                       {"warehouse", "hdfs://HDFS1006531/data/paimon1"},
```
Comment (wsjz) [inline be/src/vec/exec/scan/paimon_reader.cpp:62]: and here

**Comment by yuxuan-luo** [comment]
https://github.com/apache/doris/pull/19681 is open

**Pull request closed without merging**